src/main/java/com/habicus/core/controller/v1/goal/GoalController.java

```diff
@@ -61,7 +72,101 @@ public void setUserService(UserService userService) {
+    } catch (Exception e) {
+      responseData =
+          ControllerUtil.createResponse(
+              "Failure to remove specified goal.", HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    return new ResponseEntity<>(responseData, HttpStatus.CREATED);
+  }
+
+  /**
+   * Creates a user goal and then persists it for that specific user Grabs the current security
+   * subject to derive the proper user
+   *
+   * @return {@link ResponseEntity} of the {@link com.habicus.core.model.User} {@link Goal}
+   */
+  @PostMapping("/goal")
+  public ResponseEntity<ImmutableMap> createUserGoal(
+      @RequestBody @Valid Goal goal, BindingResult bindResult, Principal principal) {
+    ImmutableMap<String, String> responseData;
+
+    if (bindResult.hasErrors()) {
+      // TODO: Create error response class to log the bindResult errors
+      LOGGER.info("Creation failure during object binding: " + goal.toString());
+
+      responseData =
+          ControllerUtil.createResponse(
+              "Failure to create new Goal", goal.toString(), HttpStatus.INTERNAL_SERVER_ERROR);
+      return new ResponseEntity<>(responseData, HttpStatus.CREATED);
+    }
+
+    Optional<Goal> createdGoal = goalService.addNewGoal(principal, goal);
+    LOGGER.info("Goal created and saved successfully");
+
+    responseData =
+        ControllerUtil.createResponse(
+            "Goal saved and created successfully!",
+            createdGoal.get().toString(),
+            HttpStatus.CREATED);
+    return new ResponseEntity<>(responseData, HttpStatus.CREATED);
+  }
+
+  @PutMapping("/goal/{goalId}")
+  public ResponseEntity<ImmutableMap<String, String>> updateUserGoal(
+      @PathVariable("goalId") int goalId,
+      @Valid @RequestBody Goal goal,
+      BindingResult bindResult,
+      Principal principal)
+      throws NoGoalsFoundException {
+
+    ImmutableMap<String, String> responseData;
+    if (bindResult.hasErrors()) {
+      // TODO: Create error response class to log the bindResult errors
+      LOGGER.info("Error binding input goal object: " + goal.toString());
+
+      responseData =
+          ControllerUtil.createResponse(
+              "Failure to update goal", goal.toString(), HttpStatus.INTERNAL_SERVER_ERROR);
+      return new ResponseEntity<>(responseData, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+
+    Optional<Goal> updatedGoal = goalService.updateExistingGoal(principal, goal);
+    LOGGER.info("Goal updated successfully");
+
+    try {
+      responseData =
+          ControllerUtil.createResponse(
+              "Goal updated successfully", updatedGoal.get().toString(), HttpStatus.OK);
+      return new ResponseEntity<>(responseData, HttpStatus.OK);
+    } catch (StandardUserException | StandardGoalException e) {
+      responseData =
+          ControllerUtil.createResponse(
+              "Permissions Exception Updating Goal", e.toString(), HttpStatus.OK);
+      return new ResponseEntity<>(responseData, HttpStatus.OK);
```

Comment:
If we try-catch the actual update portion, I think it makes more sense for this to return a server exception if we actually fail on updating the goal.. otherwise, we probably do want to indeed return a success response